CRM/Core/BAO/Phone.php

```diff
@@ -92,7 +92,8 @@ public static function allPhones($id, $updateBlankLocInfo = FALSE, $type = NULL,
 
     $cond = NULL;
     if ($type) {
-      $phoneTypeId = array_search($type, CRM_Core_PseudoConstant::get('CRM_Core_DAO_Phone', 'phone_type_id'));
+      $phoneTypes = CRM_Core_OptionGroup::values('phone_type', TRUE, FALSE, FALSE, NULL, 'name');
+      $phoneTypeId = CRM_Utils_Array::value($type, $phoneTypes);
```

Comment:
I'm not sure what the latest standard is but I think it would be preferred to replace both these lines with `$phoneTypeId = CRM_Core_PseudoConstant::getKey('CRM_Core_DAO_Phone', 'phone_type_id', $type)`. In any case `CRM_Utils_Array::value` is discouraged in places where you can use `$a[$b] ?? NULL` instead.

Reply:
Yes - 

```
$phoneTypeId = CRM_Core_PseudoConstant::getKey('CRM_Core_DAO_Phone', 'phone_type_id', $type);
```

is the preferred syntax for this